eng/Versions.props

```diff
@@ -31,7 +31,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- Dependencies from https://github.com/microsoft/msbuild -->
-    <MicrosoftBuildPackageVersion>16.3.0</MicrosoftBuildPackageVersion>
+    <MicrosoftBuildPackageVersion>16.4.0</MicrosoftBuildPackageVersion>
```

Comment:
Do we really need this update here?

Reply:
Yes. or test will fail due to a msbuild type mismatch